### ISSUE TITLE
[e2e] e2e framework import-restrictions need to be fixed

### ISSUE THREAD

**Issue by alejandrox1** [kubernetes/kubernetes#81803]
We recently discovered in https://github.com/kubernetes/kubernetes/pull/81246#issuecomment-523510399 that the import restrictions for the e2e testing framework, [/test/e2e/framework/.import-restrictions](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/.import-restrictions), are not being taken into account and are not correct.

The CI process uses [./hack/verify-import-boss.sh](https://github.com/kubernetes/kubernetes/blob/master/hack/verify-import-boss.sh) to verify the `.import-restrictions` files throughout k/k.
However, we didn't enable [import-boss](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/code-generator/cmd/import-boss/main.go#L79-L83) to verify imports in the e2e framework.

Furthermore, if one does enable import-boss to use the import-restrictions file for the e2e framework the process fails with the following output:

```
errors in package "k8s.io/kubernetes/test/e2e/framework/lifecycle":
the following imports did not match any allowed prefix:
  k8s.io/kubernetes/pkg/version
  k8s.io/kubernetes/test/e2e/framework
  k8s.io/kubernetes/test/e2e/framework/config
  k8s.io/kubernetes/test/e2e/framework/kubelet
  k8s.io/kubernetes/test/e2e/framework/psp
  k8s.io/utils/io


errors in package "k8s.io/kubernetes/test/e2e/framework/providers/vsphere":
the following imports did not match any allowed prefix:
  k8s.io/kubernetes/pkg/version
  k8s.io/kubernetes/test/e2e/framework
  k8s.io/kubernetes/test/e2e/framework/config
  k8s.io/kubernetes/test/e2e/framework/kubelet
  k8s.io/kubernetes/test/e2e/framework/psp
  k8s.io/utils/io


errors in package "k8s.io/kubernetes/test/e2e/framework/job":
the following imports did not match any allowed prefix:
  k8s.io/kubernetes/pkg/version
  k8s.io/kubernetes/test/e2e/framework
  k8s.io/kubernetes/test/e2e/framework/config
  k8s.io/kubernetes/test/e2e/framework/kubelet
  k8s.io/kubernetes/test/e2e/framework/psp
  k8s.io/utils/io


errors in package "k8s.io/kubernetes/test/e2e/framework/providers/gce":
the following imports did not match any allowed prefix:
  k8s.io/kubernetes/pkg/version
  k8s.io/kubernetes/test/e2e/framework
  k8s.io/kubernetes/test/e2e/framework/config
  k8s.io/kubernetes/test/e2e/framework/kubelet
  k8s.io/kubernetes/test/e2e/framework/psp
  k8s.io/kubernetes/test/e2e/framework/service
  k8s.io/utils/io


errors in package "k8s.io/kubernetes/test/e2e/framework/gpu":
the following imports did not match any allowed prefix:
  k8s.io/kubernetes/pkg/version
  k8s.io/kubernetes/test/e2e/framework
  k8s.io/kubernetes/test/e2e/framework/config
  k8s.io/kubernetes/test/e2e/framework/kubelet
  k8s.io/kubernetes/test/e2e/framework/psp
  k8s.io/utils/io


errors in package "k8s.io/kubernetes/test/e2e/framework/service":
the following imports did not match any allowed prefix:
  k8s.io/kubernetes/pkg/version
  k8s.io/kubernetes/test/e2e/framework
  k8s.io/kubernetes/test/e2e/framework/config
  k8s.io/kubernetes/test/e2e/framework/kubelet
  k8s.io/kubernetes/test/e2e/framework/psp
  k8s.io/utils/io


errors in package "k8s.io/kubernetes/test/e2e/framework/providers/aws":
the following imports did not match any allowed prefix:
  k8s.io/kubernetes/pkg/version
  k8s.io/kubernetes/test/e2e/framework
  k8s.io/kubernetes/test/e2e/framework/config
  k8s.io/kubernetes/test/e2e/framework/kubelet
  k8s.io/kubernetes/test/e2e/framework/psp
  k8s.io/utils/io


errors in package "k8s.io/kubernetes/test/e2e/framework/timer":
the following imports did not match any allowed prefix:
  k8s.io/utils/io


errors in package "k8s.io/kubernetes/test/e2e/framework/volume":
the following imports did not match any allowed prefix:
  k8s.io/kubernetes/pkg/version
  k8s.io/kubernetes/test/e2e/framework
  k8s.io/kubernetes/test/e2e/framework/config
  k8s.io/kubernetes/test/e2e/framework/kubelet
  k8s.io/kubernetes/test/e2e/framework/psp
  k8s.io/kubernetes/test/e2e/storage/utils
  k8s.io/utils/io


errors in package "k8s.io/kubernetes/test/e2e/framework/providers/kubemark":
the following imports did not match any allowed prefix:
  k8s.io/kubernetes/pkg/api/v1/resource
  k8s.io/kubernetes/pkg/api/v1/service
  k8s.io/kubernetes/pkg/cloudprovider/providers
  k8s.io/kubernetes/pkg/cloudprovider/providers/openstack
  k8s.io/kubernetes/pkg/credentialprovider
  k8s.io/kubernetes/pkg/credentialprovider/aws
  k8s.io/kubernetes/pkg/credentialprovider/azure
  k8s.io/kubernetes/pkg/credentialprovider/gcp
  k8s.io/kubernetes/pkg/credentialprovider/secrets
  k8s.io/kubernetes/pkg/kubelet
  k8s.io/kubernetes/pkg/kubelet/cadvisor
  k8s.io/kubernetes/pkg/kubelet/certificate
  k8s.io/kubernetes/pkg/kubelet/certificate/bootstrap
  k8s.io/kubernetes/pkg/kubelet/cloudresource
  k8s.io/kubernetes/pkg/kubelet/cm
  k8s.io/kubernetes/pkg/kubelet/cm/cpumanager
  k8s.io/kubernetes/pkg/kubelet/cm/cpumanager/state
  k8s.io/kubernetes/pkg/kubelet/cm/cpumanager/topology
  k8s.io/kubernetes/pkg/kubelet/cm/cpuset
  k8s.io/kubernetes/pkg/kubelet/cm/devicemanager
  k8s.io/kubernetes/pkg/kubelet/cm/devicemanager/checkpoint
  k8s.io/kubernetes/pkg/kubelet/cm/topologymanager
  k8s.io/kubernetes/pkg/kubelet/cm/topologymanager/socketmask
  k8s.io/kubernetes/pkg/kubelet/cm/util
  k8s.io/kubernetes/pkg/kubelet/config
  k8s.io/kubernetes/pkg/kubelet/dockershim
  k8s.io/kubernetes/pkg/kubelet/dockershim/cm
  k8s.io/kubernetes/pkg/kubelet/dockershim/libdocker
  k8s.io/kubernetes/pkg/kubelet/dockershim/network
  k8s.io/kubernetes/pkg/kubelet/dockershim/network/cni
  k8s.io/kubernetes/pkg/kubelet/dockershim/network/hostport
  k8s.io/kubernetes/pkg/kubelet/dockershim/network/kubenet
  k8s.io/kubernetes/pkg/kubelet/dockershim/network/metrics
  k8s.io/kubernetes/pkg/kubelet/dockershim/remote
  k8s.io/kubernetes/pkg/kubelet/envvars
  k8s.io/kubernetes/pkg/kubelet/eviction
  k8s.io/kubernetes/pkg/kubelet/eviction/api
  k8s.io/kubernetes/pkg/kubelet/images
  k8s.io/kubernetes/pkg/kubelet/kubeletconfig
  k8s.io/kubernetes/pkg/kubelet/kubeletconfig/checkpoint
  k8s.io/kubernetes/pkg/kubelet/kubeletconfig/checkpoint/store
  k8s.io/kubernetes/pkg/kubelet/kubeletconfig/configfiles
  k8s.io/kubernetes/pkg/kubelet/kubeletconfig/status
  k8s.io/kubernetes/pkg/kubelet/kubeletconfig/util/codec
  k8s.io/kubernetes/pkg/kubelet/kubeletconfig/util/files
  k8s.io/kubernetes/pkg/kubelet/kubeletconfig/util/log
  k8s.io/kubernetes/pkg/kubelet/kubeletconfig/util/panic
  k8s.io/kubernetes/pkg/kubelet/kuberuntime
  k8s.io/kubernetes/pkg/kubelet/kuberuntime/logs
  k8s.io/kubernetes/pkg/kubelet/leaky
  k8s.io/kubernetes/pkg/kubelet/logs
  k8s.io/kubernetes/pkg/kubelet/mountpod
  k8s.io/kubernetes/pkg/kubelet/network/dns
  k8s.io/kubernetes/pkg/kubelet/nodelease
  k8s.io/kubernetes/pkg/kubelet/nodestatus
  k8s.io/kubernetes/pkg/kubelet/oom
  k8s.io/kubernetes/pkg/kubelet/pleg
  k8s.io/kubernetes/pkg/kubelet/pluginmanager
  k8s.io/kubernetes/pkg/kubelet/pluginmanager/cache
  k8s.io/kubernetes/pkg/kubelet/pluginmanager/metrics
  k8s.io/kubernetes/pkg/kubelet/pluginmanager/operationexecutor
  k8s.io/kubernetes/pkg/kubelet/pluginmanager/pluginwatcher
  k8s.io/kubernetes/pkg/kubelet/pluginmanager/pluginwatcher/example_plugin_apis/v1beta1
  k8s.io/kubernetes/pkg/kubelet/pluginmanager/pluginwatcher/example_plugin_apis/v1beta2
  k8s.io/kubernetes/pkg/kubelet/pluginmanager/reconciler
  k8s.io/kubernetes/pkg/kubelet/preemption
  k8s.io/kubernetes/pkg/kubelet/prober
  k8s.io/kubernetes/pkg/kubelet/prober/results
  k8s.io/kubernetes/pkg/kubelet/qos
  k8s.io/kubernetes/pkg/kubelet/remote
  k8s.io/kubernetes/pkg/kubelet/runtimeclass
  k8s.io/kubernetes/pkg/kubelet/server
  k8s.io/kubernetes/pkg/kubelet/server/metrics
  k8s.io/kubernetes/pkg/kubelet/server/portforward
  k8s.io/kubernetes/pkg/kubelet/server/remotecommand
  k8s.io/kubernetes/pkg/kubelet/server/stats
  k8s.io/kubernetes/pkg/kubelet/server/streaming
  k8s.io/kubernetes/pkg/kubelet/stats
  k8s.io/kubernetes/pkg/kubelet/stats/pidlimit
  k8s.io/kubernetes/pkg/kubelet/status
  k8s.io/kubernetes/pkg/kubelet/token
  k8s.io/kubernetes/pkg/kubelet/volumemanager
  k8s.io/kubernetes/pkg/kubelet/volumemanager/cache
  k8s.io/kubernetes/pkg/kubelet/volumemanager/metrics
  k8s.io/kubernetes/pkg/kubelet/volumemanager/populator
  k8s.io/kubernetes/pkg/kubelet/volumemanager/reconciler
  k8s.io/kubernetes/pkg/kubemark
  k8s.io/kubernetes/pkg/probe
  k8s.io/kubernetes/pkg/probe/exec
  k8s.io/kubernetes/pkg/probe/http
  k8s.io/kubernetes/pkg/probe/tcp
  k8s.io/kubernetes/pkg/proxy
  k8s.io/kubernetes/pkg/proxy/apis
  k8s.io/kubernetes/pkg/proxy/apis/config
  k8s.io/kubernetes/pkg/proxy/apis/config/scheme
  k8s.io/kubernetes/pkg/proxy/apis/config/v1alpha1
  k8s.io/kubernetes/pkg/proxy/apis/config/validation
  k8s.io/kubernetes/pkg/proxy/config
  k8s.io/kubernetes/pkg/proxy/healthcheck
  k8s.io/kubernetes/pkg/proxy/iptables
  k8s.io/kubernetes/pkg/proxy/ipvs
  k8s.io/kubernetes/pkg/proxy/metrics
  k8s.io/kubernetes/pkg/proxy/userspace
  k8s.io/kubernetes/pkg/proxy/util
  k8s.io/kubernetes/pkg/security/podsecuritypolicy/sysctl
  k8s.io/kubernetes/pkg/securitycontext
  k8s.io/kubernetes/pkg/util/async
  k8s.io/kubernetes/pkg/util/bandwidth
  k8s.io/kubernetes/pkg/util/config
  k8s.io/kubernetes/pkg/util/configz
  k8s.io/kubernetes/pkg/util/conntrack
  k8s.io/kubernetes/pkg/util/dbus
  k8s.io/kubernetes/pkg/util/ebtables
  k8s.io/kubernetes/pkg/util/env
  k8s.io/kubernetes/pkg/util/flag
  k8s.io/kubernetes/pkg/util/flock
  k8s.io/kubernetes/pkg/util/goroutinemap
  k8s.io/kubernetes/pkg/util/goroutinemap/exponentialbackoff
  k8s.io/kubernetes/pkg/util/ipset
  k8s.io/kubernetes/pkg/util/iptables
  k8s.io/kubernetes/pkg/util/ipvs
  k8s.io/kubernetes/pkg/util/oom
  k8s.io/kubernetes/pkg/util/pod
  k8s.io/kubernetes/pkg/util/procfs
  k8s.io/kubernetes/pkg/util/removeall
  k8s.io/kubernetes/pkg/util/rlimit
  k8s.io/kubernetes/pkg/util/selinux
  k8s.io/kubernetes/pkg/util/sysctl
  k8s.io/kubernetes/pkg/util/tail
  k8s.io/kubernetes/pkg/version
  k8s.io/kubernetes/pkg/version/verflag
  k8s.io/kubernetes/test/e2e/framework
  k8s.io/kubernetes/test/e2e/framework/config
  k8s.io/kubernetes/test/e2e/framework/kubelet
  k8s.io/kubernetes/test/e2e/framework/psp
  k8s.io/utils/inotify
  k8s.io/utils/io
  k8s.io/utils/keymutex


errors in package "k8s.io/kubernetes/test/e2e/framework/kubelet":
the following imports did not match any allowed prefix:
  k8s.io/utils/io


errors in package "k8s.io/kubernetes/test/e2e/framework/ingress":
the following imports did not match any allowed prefix:
  k8s.io/kubernetes/pkg/version
  k8s.io/kubernetes/test/e2e/framework
  k8s.io/kubernetes/test/e2e/framework/config
  k8s.io/kubernetes/test/e2e/framework/kubelet
  k8s.io/kubernetes/test/e2e/framework/psp
  k8s.io/kubernetes/test/e2e/framework/service
  k8s.io/utils/io


errors in package "k8s.io/kubernetes/test/e2e/framework/providers/azure":
the following imports did not match any allowed prefix:
  k8s.io/kubernetes/pkg/version
  k8s.io/kubernetes/test/e2e/framework
  k8s.io/kubernetes/test/e2e/framework/config
  k8s.io/kubernetes/test/e2e/framework/kubelet
  k8s.io/kubernetes/test/e2e/framework/psp
  k8s.io/utils/io


errors in package "k8s.io/kubernetes/test/e2e/framework/metrics":
the following imports did not match any allowed prefix:
  k8s.io/utils/io


errors in package "k8s.io/kubernetes/test/e2e/framework/perf":
the following imports did not match any allowed prefix:
  k8s.io/kubernetes/test/e2e/framework/kubelet
  k8s.io/utils/io


errors in package "k8s.io/kubernetes/test/e2e/framework/providers/openstack":
the following imports did not match any allowed prefix:
  k8s.io/kubernetes/pkg/version
  k8s.io/kubernetes/test/e2e/framework
  k8s.io/kubernetes/test/e2e/framework/config
  k8s.io/kubernetes/test/e2e/framework/kubelet
  k8s.io/kubernetes/test/e2e/framework/psp
  k8s.io/utils/io


errors in package "k8s.io/kubernetes/test/e2e/framework/node":
the following imports did not match any allowed prefix:
  k8s.io/utils/io


errors in package "k8s.io/kubernetes/test/e2e/framework/pod":
the following imports did not match any allowed prefix:
  k8s.io/utils/io


errors in package "k8s.io/kubernetes/test/e2e/framework/statefulset":
the following imports did not match any allowed prefix:
  k8s.io/kubernetes/pkg/version
  k8s.io/kubernetes/test/e2e/framework
  k8s.io/kubernetes/test/e2e/framework/config
  k8s.io/kubernetes/test/e2e/framework/kubelet
  k8s.io/kubernetes/test/e2e/framework/psp
  k8s.io/utils/io


errors in package "k8s.io/kubernetes/test/e2e/framework/replicaset":
the following imports did not match any allowed prefix:
  k8s.io/kubernetes/pkg/version
  k8s.io/kubernetes/test/e2e/framework
  k8s.io/kubernetes/test/e2e/framework/config
  k8s.io/kubernetes/test/e2e/framework/kubelet
  k8s.io/kubernetes/test/e2e/framework/psp
  k8s.io/utils/io


!!! Error in ./hack/verify-import-boss.sh:28
  Error in ./hack/verify-import-boss.sh:28. '$(kube::util::find-binary "import-boss") --verify-only' exited with status 1
Call stack:
  1: ./hack/verify-import-boss.sh:28 main(...)
Exiting with status 1
```

We need to:
1. Enable import-boss to verify import restrictions within the e2e framework by adding `"k8s.io/kubernetes/test/e2e/framework/..."` to `arguments.InputDirs`.
2. Fix [/test/e2e/framework/.import-restrictions](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/.import-restrictions), based on the output

/area e2e-test-framework
/sig testing
/kind bug
/priority important-soon
/cc @andrewsykim @oomichi

**Comment by oomichi** [comment]
I think this still is important for us.

/lifecycle active

**Comment by gavinfish** [comment]
I think issue should be fixed by #87265.

**Comment by oomichi** [comment]
Oops, I forgot this issue and created a new one (https://github.com/kubernetes/kubernetes/issues/87264) instead.
Yes, this should be closed with https://github.com/kubernetes/kubernetes/pull/87265

/close
